source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -998,6 +998,337 @@ See the `Load Balancer Specification <../load-balancers/load-balancers.rst#event
       connectionId: number;
     }
 
+Connection Pool Logging
+~~~~~~~~~~~~~~~~~~~~~~~
+Please refer to the `logging specification <../logging/logging.rst>`_ for details on logging implementations in general, including log levels, log
+components, handling of null values in log messages, and structured versus unstructured logging.
+
+Drivers MUST support logging of connection pool information via the following types of log messages. These messages MUST be logged at ``Debug`` level
+and use the ``connection`` log component. These messages MUST be emitted when specified in “Connection Pool Behaviors”.
+
+The log messages are intended to match the information contained in the events above. Drivers MAY implement connection logging support via an event
+subscriber if it is convenient to do so.
+
+The types used in the structured message definitions below are demonstrative, and drivers MAY use similar types instead so long as the information
+is present (e.g. a double instead of an integer, or a string instead of an integer if the structured logging framework does not support numeric types.)
+
+Common Fields
+-------------
+All connection log messages MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - serverHost
+     - String
+     -  the hostname or IP address for the endpoint the pool is for.
+
+   * - serverPort
+     - Int
+     - The port for the endpoint the pool is for. Optional; only present if a port was specified.
```

Comment:
this seems reasonable in cases where there is a port. but do ports actually have any significance for unix domain sockets? from looking at the libmongoc code, [`mongoc_client_connect_unix`](https://github.com/mongodb/mongo-c-driver/blob/b6dad0ccb4ba2e1dfd3f4453a01ec440487af563/src/libmongoc/src/mongoc/mongoc-client.c#L743-L790) does nothing with the port (vs [`mongoc_client_connect_tcp`](https://github.com/mongodb/mongo-c-driver/blob/b6dad0ccb4ba2e1dfd3f4453a01ec440487af563/src/libmongoc/src/mongoc/mongoc-client.c#L645-L722) which uses `host->port`.)

Reply:
Yeah, in that case I don't think there would be one. I believe our representation of Unix domain sockets is just a path to a file on disk.

On a related note: would we want to combine these two host/port fields into a single "address" field? Might be a little bit cleaner since as you point out ports only apply to certain types of addresses.

Reply:
I'm fine with adding language saying drivers should put the default port here when it is being used. that said, I'm not sure we can do anything more than say drivers should do it, because I don't see how we can write a test for it 🤔 the connection strings we get from drivers-evergreen-tools always do have ports specified, and there's no way from a unified test to override the full connection string - only particular options on it. so without prose testing every log message you can't really confirm drivers do this right. a SHOULD might be good enough though. 

> On a related note: would we want to combine these two host/port fields into a single "address" field? Might be a little bit cleaner since as you point out ports only apply to certain types of addresses.

when there is no port (i.e. Unix domain socket) drivers who log strings can just emit the "with port X" portion and drivers doing structured logging can just not attach a port value at all, so I think we end up in a similar place. if you feel strongly about this I'm not deeply opposed to it, but if it's only a suggestion I think I'd rather stick with the precedent we set in the command log messages of keeping these fields separate. also, while I don't think we should let testability leak much into public API decisions, it might be worth noting that combining the fields would also prevent us from actually testing that drivers include both the host and port and that they are types we expect them to be (string and number) - all we could do is assert that an address string is present.

Reply:
I updated the language here:
> The port for the endpoint the pool is for. Optional; not present for Unix domain sockets. When
       the user does not specify a port and the default (27017) is used, the driver SHOULD include it here. 

going to make a follow up PR to do similar for the command log messages